File: mobile-engage/src/main/java/com/emarsys/mobileengage/push/PushInternal.java
package com.emarsys.mobileengage.push;

import android.content.Intent;

import com.emarsys.core.api.result.CompletionListener;
import com.emarsys.mobileengage.api.event.EventHandler;
import com.emarsys.mobileengage.api.push.NotificationInformationListener;

public interface PushInternal {
    void setPushToken(String pushToken, CompletionListener completionListener);

    String getPushToken();

    void clearPushToken(CompletionListener completionListener);

    void trackMessageOpen(Intent intent, CompletionListener completionListener);

    void setNotificationEventHandler(EventHandler notificationEventHandler);

    void setSilentMessageEventHandler(EventHandler silentMessageEventHandler);

    void setNotificationInformationListener(NotificationInformationListener notificationInformationListener);

    void setSilentNotificationInformationListener(NotificationInformationListener silentNotificationInformationListener);
}

File: mobile-engage/src/main/java/com/emarsys/mobileengage/push/PushInternal.kt
package com.emarsys.mobileengage.push

import com.emarsys.core.api.result.CompletionListener
import com.emarsys.mobileengage.api.event.EventHandler
import com.emarsys.mobileengage.api.push.NotificationInformationListener

interface PushInternal {
    fun setPushToken(pushToken: String, completionListener: CompletionListener?)
    val pushToken: String?
    fun clearPushToken(completionListener: CompletionListener?)
    fun trackMessageOpen(sid: String?, completionListener: CompletionListener?)
    fun setNotificationEventHandler(notificationEventHandler: EventHandler)
    fun setSilentMessageEventHandler(silentMessageEventHandler: EventHandler)
    fun setNotificationInformationListener(notificationInformationListener: NotificationInformationListener)
    fun setSilentNotificationInformationListener(silentNotificationInformationListener: NotificationInformationListener)
}